Sources/GoogleDriveClient/ExecuteScript.swift

```
@@ -0,0 +1,106 @@
+    public var function: String
+    public var args: [String]
+  }
+
+  public enum Error: Swift.Error, Sendable, Equatable {
+    case notAuthorized
+    case response(statusCode: Int?, data: Data)
+  }
+
+  public typealias Run = @Sendable (Params) async throws -> Data
+
+  public init(run: @escaping Run) {
+    self.run = run
+  }
+
+  public var run: Run
+
+  public func callAsFunction(_ params: Params) async throws -> Data {
+    try await run(params)
+  }
+
+  public func callAsFunction(
+    scriptId: String,
+    function: String,
+    args: [String] = []
+  ) async throws -> Data {
+    try await run(.init(
+      scriptId: scriptId,
+      function: function,
+      args: args
+    ))
+  }
+}
+
+extension ExecuteScript {
+  public static func live(
+    auth: Auth,
+    keychain: Keychain,
+    httpClient: HTTPClient
+  ) -> ExecuteScript {
+    ExecuteScript { params in
+      try await auth.refreshToken()
+
+      guard let credentials = await keychain.loadCredentials() else {
+        throw Error.notAuthorized
+      }
+
+      let request: URLRequest = {
+        var components = URLComponents()
+        components.scheme = "https"
+        components.host = "script.googleapis.com"
+        components.path = "/v1/scripts/\(params.scriptId):run"
+        print("Components: \(components)")
+
+        var request = URLRequest(url: components.url!)
+        request.httpMethod = "POST"
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.setValue(
+          "\(credentials.tokenType) \(credentials.accessToken)",
+          forHTTPHeaderField: "Authorization"
+        )
+        print("Authorization: \(credentials.tokenType) \(credentials.accessToken)")
+
+        let requestBody: [String: Any] = [
+          "function": params.function,
+          "parameters": params.args,
+          "devMode": false
+        ]
+        request.httpBody = try? JSONSerialization.data(withJSONObject: requestBody)
```

We should not ignore the error with `try?`. If the request body can't be encoded, we should throw the error.

```swift
let request: URLRequest = try {
  ...
  request.httpBody = try JSONSerialization.data(withJSONObject: requestBody)
  ...
}
```

---

Agree. Removed "?" to propagate error.